src/classes/CDL_CascadeDeleteLookups.cls

```diff
@@ -0,0 +1,350 @@
+/*
+    Copyright (c) 2017, Salesforce.org
+    All rights reserved.
+    
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+ * @author Salesforce.org
+ * @date 2017
+ * @group Cascade Delete
+ * @description Cascade deletion and undeletion Domain Classes
+ */
+
+public class CDL_CascadeDeleteLookups {
+
+    @TestVisible
+    private static final String HTML_TEXT_CHILD_RECORDS_FORMAT = '<li><a href="#" onclick="window.open(\'\'/{0}\'\', \'\'\'\');return false;" target="_blank">{1}</a></li>';
+
+    /** @description The single instance of the CascadeDelete. */
+    private static CascadeDelete cascadeDelete;
+
+    /*******************************************************************************************************
+    * @description Gets the current user Theme Context (Lightning Experience or Classic).
+    * @return Boolean.
+    ********************************************************************************************************/
+    public static Boolean isLightningContext {
+        get {
+            return (isLightningContext == null)
```

Comment:
The var `isLightningContext` is not assigned, so this check will be done every time. Feel free to break it down, ie does not have to be as indicated in my comment. So, either is fine:

```
get {
    return isLightningContext = isLightningContext == null
        ? USER_UserService.isUserRunningLightning() || USER_UserService.isUserRunningMobile()
        : isLightningContext;
}

```
OR

```
get {
    if (isLightningContext == null) {
        isLightningContext = USER_UserService.isUserRunningLightning() || USER_UserService.isUserRunningMobile();
    }
    return isLightningContext;
}
```